src/Microsoft.PowerShell.Commands.Utility/commands/utility/WebCmdlet/JsonObject.cs

```diff
@@ -42,11 +42,38 @@ public static object ConvertFromJson(string input, out ErrorRecord error)
             {
                 throw new ArgumentNullException("input");
             }
+
             error = null;
 #if CORECLR
             object obj = null;
             try
             {
+                // JsonConvert.DeserializeObject does not throw an exception when an invalid array is passed,
+                // so we need to do extra logic to identify this scenario using JArray.Parse(input).
+
+                // If input contains '[' or ']', we would like to analyze it further as these markers for a Json data array.
+                if ((input.IndexOf("[", StringComparison.OrdinalIgnoreCase) >= 0) ||
+                    (input.IndexOf("]", StringComparison.OrdinalIgnoreCase) >= 0))
+                {
+                    // Remove carriage return, new-line and empty space from the string.
+                    string stringToValidate = input.Replace(System.Environment.NewLine, string.Empty);
```

Comment:
The comment doesn't match the code. Carriage return is `\r`, but `Environemnt.NewLine` is `\r\n` on windows and `\n` on Unix. And what if the json string is has `\r\n` but we are calling `ConvertFrom-Json` on Unix?